internal/pkg/cachecli/cache_clean.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2017-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cachecli
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/sylabs/singularity/internal/pkg/client/cache"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+// CleanLibraryCache : clean only library cache, will return a error if one occurs
+func CleanLibraryCache() error {
+	sylog.Debugf("Removing: %v", cache.Library())
+
+	err := os.RemoveAll(cache.Library())
+
+	return err
+}
+
+// CleanOciCache : clean only oci cache, will return a error if one occurs
+func CleanOciCache() error {
+	sylog.Debugf("Removing: %v", cache.OciTemp())
+
+	err := os.RemoveAll(cache.OciTemp())
+
+	return err
+}
+
+// CleanBlobCache : clean only blob cache, will return a error if one occurs
+func CleanBlobCache() error {
+	sylog.Debugf("Removing: %v", cache.OciBlob())
+
+	err := os.RemoveAll(cache.OciBlob())
+
+	return err
+
+}
+
+func cleanLibraryCache(cacheName string) bool {
+	foundMatch := false
+	libraryCacheFiles, err := ioutil.ReadDir(cache.Library())
+	if err != nil {
+		sylog.Fatalf("Failed while opening cache folder: %v", err)
```

Comment:
Generally it's a better practice to return an `error` to be handled by the caller instead of calling `Sylog.Fatalf` since it gives the calling function the chance to handle the error how it wants. Eg. ignoring it, performing some kind of cleanup, etc.

Reply:
Ahh, That makes more sense now that you explained it, I first thought it was weird passing error from function to function, but it makes sense now :+1: